src/bobtail/bobtail_miner.cpp

```diff
@@ -113,17 +113,26 @@ uint64_t BobtailBlockAssembler::reserveBlockSize(const CScript &scriptPubKeyIn,
     return nHeaderSize + nCoinbaseSize;
 }
 
-CTransactionRef BobtailBlockAssembler::coinbaseTx(const CScript &scriptPubKeyIn, int _nHeight, CAmount nValue)
+CTransactionRef BobtailBlockAssembler::coinbaseTx(const CScript &scriptPubKeyIn, int _nHeight, CAmount nValue, const std::set<CDagNode> &dag)
 {
     //TODO: Should payout to lowest k subblock miners
     CMutableTransaction tx;
 
     tx.vin.resize(1);
     tx.vin[0].prevout.SetNull();
-    tx.vout.resize(1);
-    tx.vout[0].scriptPubKey = scriptPubKeyIn;
-    tx.vout[0].nValue = nValue;
     tx.vin[0].scriptSig = CScript() << _nHeight << OP_0;
+    // set the vout to be bobtail K at least
+    tx.vout.resize(BOBTAIL_K);
+    CAmount valuePer = nValue / BOBTAIL_K;
+    unsigned int i = 0;
+    std::set<CDagNode>::iterator iter = dag.begin();
```

Comment:
Just making sure: the iterator is ordered by ascending hash value right? I see your comparison operator below, but I'm not 100% sure how it gets picked up by the iterator.